### ISSUE TITLE
Errors  while selecting in visual mode

### ISSUE THREAD

**Issue by rad4day** [braxtons12/blame_line.nvim#1]
Hello currently I am running into some weird errors when I select multiple lines in visual mode. Currently, it occurs whenever I try to visually select over an empty line.
Update: Also happens on non-empty lines when selecting upward not downwards.

```
Error detected while processing CursorMoved Autocommands for "*":
Error executing lua callback: ...im/plugged/blame_line.nvim/lua/blame_line/type_check.lua:66: [blame_line.nvim] invalid type given for
 parameter line in blame_line.parse_commit_committer_or_author_line
stack traceback:
        [C]: in function 'assert'
        ...im/plugged/blame_line.nvim/lua/blame_line/type_check.lua:66: in function 'check'
        ...day/.vim/plugged/blame_line.nvim/lua/blame_line/init.lua:273: in function 'parse_commit_committer_or_author_line'
        ...day/.vim/plugged/blame_line.nvim/lua/blame_line/init.lua:467: in function 'get_commit_data'
        ...day/.vim/plugged/blame_line.nvim/lua/blame_line/init.lua:553: in function 'process_blame_line'
        ...day/.vim/plugged/blame_line.nvim/lua/blame_line/init.lua:569: in function 'show'
        ...day/.vim/plugged/blame_line.nvim/lua/blame_line/init.lua:703: in function 'refresh'
        ...tfiles/vim/plugged/blame_line.nvim/plugin/blame_line.lua:71: in function <...tfiles/vim/plugged/blame_line.nvim/plugin/blam
e_line.lua:70>
```

Using nvim 0.8

```
:version
NVIM v0.8.0-dev+503-g274609a109
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/cc -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions     -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=for
mat-security     -fstack-clash-protection -fcf-protection -Wp,-U_FORTIFY_SOURCE -Wp,-D_FORTIFY_SOURCE=1 -DNVIM_TS_HAS_SET_MATCH_LIMIT
-DNVIM_TS_HAS_SET_ALLOCATOR -O2 -g -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconv
ersion -Wdouble-promotion -Wmissing-noreturn -Wmissing-format-attribute -Wmissing-prototypes -Wimplicit-fallthrough -Wsuggest-attribut
e=pure -Wsuggest-attribute=const -Wsuggest-attribute=malloc -Wsuggest-attribute=cold -Wvla -fstack-protector-strong -fno-common -fdiag
nostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL
=3
```

Config:
```vim
" Blame
lua << EOF
require("blame_line").setup {
    -- whether the blame line should be shown in visual modes
    show_in_visual = true,

    -- whether the blame line should be shown in insert mode
    show_in_insert = true,

    -- the string to prefix the blame line with
    prefix = " ",

    -- String specifying the the blame line format.
    -- Any combination of the following specifiers, along with any additional text.
    --     - `"<author>"` - the author of the change.
    --     - `"<author-mail>"` - the email of the author.
    --     - `"<author-time>"` - the time the author made the change.
    --     - `"<committer>"` - the person who committed the change to the repository.
    --     - `"<committer-mail>"` - the email of the committer.
    --     - `"<committer-time>"` - the time the change was committed to the repository.
    --     - `"<summary>"` - the commit summary/message.
    --     - `"<commit-short>"` - short portion of the commit hash.
    --     - `"<commit-long>"` - the full commit hash.
    template = "<author> • <author-time> • <summary>",

    -- The date format settings, for `"<author-time>"` and `"<committer-time>"`
    date = {
        -- whether the date should be relative instead of precise
        -- (I.E. "3 days ago" instead of "09-06-2022".
        relative = true,

        -- `strftime` compatible format string.
        -- Only used if `date.relative == false`
        format = "%d-%m-%y",
    },

    -- The highlight group to highlight the blame line with.
    -- The highlight of this group defaults to `Comment`.
    hl_group = "BlameLineNvim",

    -- The delay in milliseconds between a cursor movement and
    -- when the blame line should appear/update
    delay = 0,
}
EOF
```

Sadly I currently do not have the spare time to debug this myself :/

**Comment by braxtons12** [comment]
Hi, thanks for the detailed report. I'll check this out asap!
Will hopefully have a fix for you sometime tomorrow

**Comment by braxtons12** [comment]
This should be fixed now in [9840596](https://github.com/braxtons12/blame_line.nvim/commit/9840596ebc85554e41051f7de0ede0622881714f)

**Comment by rad4day** [comment]
I have done some quick testing. Seems to work just perfectly.

Thank you very much :)
